pkg/api/controller.go

```diff
@@ -43,20 +43,37 @@ func (c *Controller) Run() error {
 	engine.Use(log.SessionLogger(c.Log), handlers.RecoveryHandler(handlers.RecoveryLogger(c.Log),
 		handlers.PrintRecoveryStack(false)))
 
-	c.ImageStore = storage.NewImageStore(c.Config.Storage.RootDirectory, c.Config.Storage.GC,
-		c.Config.Storage.Dedupe, c.Log)
-	if c.ImageStore == nil {
-		// we can't proceed without at least a image store
+	c.Router = engine
+	c.Router.UseEncodedPath()
+
+	// Assign global level image storage
+	c.ImageStore = make(map[string]*storage.ImageStore)
```

Comment:
make this a Trie.

Reply:
Why do we need Trie here instead of map ?

Reply:
ignore this as well.